### PR TITLE
chore(merge): 7cb376a and db60ae0

### PR DESCRIPTION
**Pull request by julien-deramond**
> **Warning**
> Must be rebased on _main_ after https://github.com/Orange-OpenSource/Orange-Boosted-Bootstrap/pull/1751 is merged

This PR is linked to https://github.com/Orange-OpenSource/Orange-Boosted-Bootstrap/pull/1751 where we'll merge two commits that could have been complicated to review in the middle of everything:
* https://github.com/twbs/bootstrap/commit/7cb376a8fd2133d28960c3a1a1fa2b5016747bae
* https://github.com/twbs/bootstrap/commit/db60ae0625ee989c8fe08076663759c287207c9a

On top of that, in the same spirit, I got rid of all `$border-width` Boosted direct calls and replaced them by the custom property.

Found out while modifying it that we forgot to remove style in our Boosted docs.

### CODE REVIEW

File: scss/_variables.scss [hunk 1]
```diff
@@ -834,8 +834,7 @@ $table-hover-bg:                        rgba($black, $table-hover-bg-factor) !de
 $table-variant-hover-bg-factor:         .2 !default; // Boosted mod
 
 $table-border-factor:                   .4 !default; // Boosted mod
-/* stylelint-disable-next-line function-disallowed-list */
-$table-border-width:                    calc(var(--#{$prefix}border-width) * .5) !default; // Boosted mod
+$table-border-width:                    calc(var(--#{$prefix}border-width) * .5) !default; /* stylelint-disable-line function-disallowed-list */ // Boosted mod
```
[the remark this answers is not in the document]
Comment (julien-deramond) [reply scss/_variables.scss:837]: Nope it doesn't work

File: scss/_button-group.scss [hunk 2]
```diff
@@ -45,7 +45,7 @@
   // Prevent double borders when buttons are next to each other
   > :not(.btn-check:first-child) + .btn,
   > .btn-group:not(:first-child) {
-    margin-left: -$btn-border-width;
+    margin-left: calc($btn-border-width * -1); // stylelint-disable-line function-disallowed-list
```
Comment (MewenLeHo) [inline scss/_button-group.scss:48]:
```suggestion
    margin-left: calc(-1 * $btn-border-width); // stylelint-disable-line function-disallowed-list
```

File: scss/_button-group.scss [hunk 3]
```diff
@@ -176,7 +176,7 @@
 
   > .btn:not(:first-child),
   > .btn-group:not(:first-child) {
-    margin-top: -$btn-border-width;
+    margin-top: calc($btn-border-width * -1); // stylelint-disable-line function-disallowed-list
```
Comment (MewenLeHo) [inline scss/_button-group.scss:179]:
```suggestion
    margin-top: calc(-1 * $btn-border-width); // stylelint-disable-line function-disallowed-list
```

File: scss/_pagination.scss [hunk 4]
```diff
@@ -110,7 +110,7 @@
     margin-left: $pagination-margin-start;
   }
 
-  @if $pagination-margin-start == ($pagination-border-width * -1) {
+  @if $pagination-margin-start == calc($pagination-border-width * -1) {
```
Comment (MewenLeHo) [inline scss/_pagination.scss:113]:
```suggestion
  @if $pagination-margin-start == calc(-1 * $pagination-border-width) {
```

File: scss/_button-group.scss (hunk 2 above)
Comment (MewenLeHo) [reply scss/_button-group.scss:48]: Possible compilation problem when -1 does not come first in calc()

File: scss/_variables.scss [hunk 5]
```diff
@@ -583,7 +583,7 @@ $border-radius-pill:          50rem !default;
 // scss-docs-end border-radius-variables
```
Comment (MewenLeHo) [inline scss/_variables.scss:583]: Any change needed from line 565 to 568?

Comment (julien-deramond) [reply scss/_variables.scss:583]: Hmm good question 🤔 I'll try if it works. Good catch! I thought it was done like this in Bootstrap but in fact they're using fixed values and not vars.

Comment (julien-deramond) [reply scss/_variables.scss:583]: Can't work because `.border-2` would be defined like this: `--bs-border-width: var(--bs-border-width);`

**Review by MewenLeHo — COMMENTED** [review]
I haven't seen any visual problem after theses changes.
I a just a bit worried by potential compilation problem because of using `calc()` with negative values. Unfortunately I don't see how it could be possible to manage it without making it unmaintenable.

**Review by MewenLeHo — APPROVED** [review]
LGTM and we will keep an eye on compilation 👀

**Pull request merged**